dfs/registry/pull.go

```diff
@@ -32,7 +32,7 @@ var (
 // Registry performs specific docker actions based on the registry index
 type Registry interface {
 	SetConnection(conn client.Connection)
-	PullImage(cancel <-chan time.Time, image string) error
+	PullImage(cancel <-chan time.Time, image string) (string, error)
```

Comment:
We are actually returning a normalized version of the `image` string that gets passed in.  It's possible that the supplied string has a host/port in it, and we don't want to return that.